### PR TITLE
pythonPackages.pushbullet: add missing dependency

### PR DESCRIPTION
**Pull request by evalexpr**
###### Motivation for this change

ZHF: #122042 

Adds missing dependency

CC @NixOS/nixos-release-managers

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122080` at cf3936f9 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pushbullet</li>
    <li>python39Packages.pushbullet</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/pushbullet/default.nix:29:3:
```
   |
29 |   meta = with lib; {
   |   ^
```
</li>
  </ul>
</details>

**Review by dotlambda — COMMENTED** [review]
Would you mind switching to `pytestCheckHook`?
Please also fix the commit message.

**Comment by evalexpr** [comment]
> Would you mind switching to `pytestCheckHook`?
> Please also fix the commit message.

Yeah I can try. Only recently learned that `pytestCheckHook` exists on the other PR, fairly new to packaging these

**Comment by dotlambda** [comment]
You will have to use
```nix
preCheck = ''
  PUSHBULLET_API_KEY=""
'';

disabledTests = [
  "test_e2e"
  "test_auth"
];
```

**Comment by evalexpr** [comment]
> ```nix
> preCheck = ''
>   export PUSHBULLET_API_KEY=""
> '';
> ```

Yup, got it

File: pkgs/development/python-modules/pushbullet/default.nix [hunk 1]
```diff
@@ -11,13 +17,16 @@ buildPythonPackage rec {
...
-
-  checkPhase = ''
-    PUSHBULLET_API_KEY="" py.test -k "not test_e2e and not test_auth"
+  preCheck = ''
+    export PUSHBULLET_API_KEY=""
   '';
+  checkInputs = [ pytestCheckHook ];
+  disabledTests = [
+    "test_e2e"
+    "test_auth"
```
Comment (dotlambda) [inline pkgs/development/python-modules/pushbullet/default.nix:28]:
```suggestion
    "test_auth_fail"
    "test_auth_success"
    "test_decryption"
```

Comment (dotlambda) [reply pkgs/development/python-modules/pushbullet/default.nix:28]: gives one more enabled test :+1:

**Pull request merged**